core/modules/wdb/QueryAction.cc

```diff
@@ -164,6 +167,9 @@ QueryAction::Impl::Impl(QueryActionArg const& a)
       _poisoned(false),
       _sendChannel(a.task->sendChannel),
       _user(a.task->user) {
+    char buf[MAX_NAME_LENGTH];
+    gethostname(buf, MAX_NAME_LENGTH);
+    _hostName = buf;
```

Comment:
If you need this to be thread-safe then it would be easier to rely on static initialization for that:

    std::string initHostName() {
            char buf[HOST_NAME_MAX+1];
            buf[HOST_NAME_MAX] = '\0';
            gethostname(buf, sizeof buf - 1);
            return std::string(buf);
    }
    std::string const& getHostname() {
        static std::string const cachedHostname = initHostName();
        return cachedHostname;
    }

And to make it super-safe I added ` - 1` to `sizeof buf`.